bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/GarbageCollectorThread.java

```diff
@@ -415,64 +446,69 @@ private void removeIfLedgerNotExists(EntryLogMetadata meta) {
                 return false;
             }
         });
+        // update entryMetadta to persistent-map
+        if (updateMetadata) {
```

Comment:
im not sure how you are deciding to update entrylogmetadata. Isn't it supposed to be updated only if there is change in the entrylogmetadata, right? In the current code, if the current argument - 'updateMetadata' value is true, isn't it going to update even if there is no change in entrylogmetadata?

Reply:
at the beginning GC creates `EntryLogMetadata` by reading and extracting entryLog file and at that time entry doesn't exist into map so, there is no need to update rocksDB. However, this flag was confusing so, I have removed the flag and updating metadata-map where it requires to be updated at ln#414.